project/src/js/ui/tournament/tournamentView.js

```diff
@@ -0,0 +1,152 @@
+
+  return layout.free(headerCtrl, bodyCtrl, empty, empty);
+}
+
+function tournamentBody(ctrl) {
+  let data = ctrl.tournament();
+  if (!data) return null;
+
+  let body = null;
+  if (data.isFinished)
+    body = tournamentContentFinished(data);
+  else if (!data.isStarted)
+    body = tournamentContentCreated(data);
+  else
+    body = tournamentContentStarted(data);
+
+  return (m('.tournamentContainer', body));
+}
+
+function tournamentContentFinished(data) {
+  return (
+    <div>
+      { tournamentHeader(data, null, null)}
+      { tournamentLeaderboard(data, true) }
+    </div>
+  );
+}
+
+function tournamentContentCreated(data) {
+  return (
+    <div>
+      { tournamentHeader(data, data.secondsToStart, 'Starts in:')}
+      { tournamentLeaderboard(data, false) }
+    </div>
+  );
+}
+
+function tournamentContentStarted(data) {
+  return (
+    <div>
+      { tournamentHeader(data, data.secondsToFinish, '')}
+      { tournamentLeaderboard(data, false) }
+      { tournamentFeaturedGame(data) }
+    </div>
+  );
+}
+
+function tournamentHeader(data, time, timeText) {
+  let variant = variantInfo(data);
+  let control = timeControl(data);
+  return (
+    <div className='basicTournamentInfo'>
+      <strong> {variant + ' • ' + control + ' • ' + data.minutes + 'M' } </strong>
+      <div className='timeInfo'>
+        <strong> {timeInfo(time, timeText)} </strong>
+      </div>
+    </div>
+  );
+}
+
+function variantInfo(data) {
+  let variant = data.variant;
+  if(variant === 'standard') {
+    if(data.schedule)
+      variant = data.schedule.speed;
+    else if(data.position)
+      variant = data.position.name;
+    else
+      variant = '';
+  }
+
+  variant = variant.split(' ')[0]; // Cut off names to first word
+
+  if (variant.length > 0)
+    variant = variant.charAt(0).toUpperCase() + variant.substring(1);
+
+  return variant;
+}
+
+function timeControl(data) {
+  let limit = (data.clock.limit / 60);
+  if (data.clock.limit === 30)
+    limit = '½';
+  else if (data.clock.limit === 45)
+    limit = '¾';
+  return limit + '+' + data.clock.increment;
+}
+
+function timeInfo(time, preceedingText) {
+  if (!time) return '';
+
+  let timeStr = '';
+  let hours = Math.floor(time / 60 / 60);
+  let mins = Math.floor(time / 60) - (hours * 60);
+  let secs = time % 60;
+  if (hours > 0)
+    timeStr = preceedingText + ' ' + hours + ':' + pad(mins, 2) + ':' + pad(secs, 2);
+  else
+    timeStr = preceedingText + ' ' + mins + ':' + pad(secs, 2);
+  return timeStr;
+}
+
+function tournamentLeaderboard(data, showTrophies) {
+  return (
+    <div className='tournamentLeaderboard'>
+      <p className='tournamentTitle'> {i18n('leaderboard')} ({data.nbPlayers} Players)</p>
+      <table className='tournamentStandings'>
+        {data.standing.players.map(createLeaderboardItemRenderer(showTrophies))}
+      </table>
+    </div>
+  );
+}
+
+function createLeaderboardItemRenderer(showTrophies) {
```

Comment:
Agree that "const podiumRank = index + 1;" is much more readable and have
made that change.



On Fri, Feb 19, 2016 at 6:10 AM, Vincent Velociter <notifications@github.com
> wrote:

> In project/src/js/ui/tournament/tournamentView.js
> <https://github.com/veloce/lichobile/pull/281#discussion_r53448545>:
>
> > +    timeStr = preceedingText + ' ' + mins + ':' + pad(secs, 2);
> > +  return timeStr;
> > +}
> > +
> > +function tournamentLeaderboard(data, showTrophies) {
> > +  return (
> > +    <div className='tournamentLeaderboard'>
> > +      <p className='tournamentTitle'> {i18n('leaderboard')} ({data.nbPlayers} Players)</p>
> > +      <table className='tournamentStandings'>
> > +        {data.standing.players.map(createLeaderboardItemRenderer(showTrophies))}
> > +      </table>
> > +    </div>
> > +  );
> > +}
> > +
> > +function createLeaderboardItemRenderer(showTrophies) {
>
> I think it should be even more simple here. Normally you don't need that
> factory. I guess you can directly write the map callback like so:
>
> renderLeaderboardItem(player, index) {
>     // better to use a var to do this than to increment the index directly
>     // it's less error prone and self explanatory
>     const podiumRank = index + 1;
>
>     ....
>     return (
>       ...
>     );
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/veloce/lichobile/pull/281/files#r53448545>.
>